starlight/starlight/core/MemMgr/MemMgr.cpp

```diff
@@ -66,20 +67,21 @@ void MemMgr::Create(uint totalSpace)
 
 }
 
-void MemMgr::Free(MemoryResource* res)
+void MemMgr::Free(uint sz, uint8_t* addr)
```

Comment:
Will the sz parameter always be equal to sizeof(*addr) where addr is the real pointer to the allocated object before you reinterpret_cast<uint8_t> it to pass into this function?

Reply:
Yes. The sz parameter should be the same value as the 'resourceSize' argument you pass into the Alloc call.

Reply:
Sweet. Would it be safe to remove a parameter (so change to `void MemMgr::Free(uint8_t* addr)`) and do `sizeof` calls on *addr within the function? Then you'd have the same one-parameter signature as C's free(...) call. Or would those `sizeof` calls fail because of the `reinterpret_cast`?

Reply:
I'll play around with the tests and let you know if sizeof *addr would yield incorrect values.

Reply:
After doing a little tinkering, I think it would be best to keep the sz parameter in the Free function solely for deallocating arrays. In dynamic allocation of arrays (something like: int* arr = new int[10]), the total size of allocated memory is not stored in the pointer itself, unlike a static allocation (int arr[10] {0}) where a sizeof call would give us the total allocated memory. 

For builtin types and classes/structs, internal sizeof calls would work fine, but arrays seem to be the limiting factor here. Let me know your thoughts.

Reply:
Makes sense, thanks for investigating - merging